src/global/reco/MatchClusters_factory.cc

```diff
@@ -27,77 +27,61 @@ namespace eicrecon {
         InitLogger(GetPrefix(), "info");
 
         m_match_algo.init(m_log);
+
+        // Set up association tags
+        for (const std::string& input_tag : GetInputTags()) {
+            // "EcalEndcapNClusters" => "EcalEndcapNClusterAssociations". Sadly we need to strip the 's' at the end
+            m_input_assoc_tags.push_back(input_tag.substr(0, input_tag.size()-1) + "Associations");
+        }
     }
 
     void MatchClusters_factory::BeginRun(const std::shared_ptr<const JEvent> &event) {
         // Nothing to do here
     }
 
     void MatchClusters_factory::Process(const std::shared_ptr<const JEvent> &event) {
-        using ClustersVector = std::vector<const edm4eic::Cluster*>;
-        using ClustersAssocVector = std::vector<const edm4eic::MCRecoClusterParticleAssociation*>;
-
-        m_log->debug("------- Process start -------");
 
         // TODO make input tags changable
         auto mc_particles = event->Get<edm4hep::MCParticle>("MCParticles");
+        auto charged_particles = event->Get<edm4eic::ReconstructedParticle>("ReconstructedChargedParticles");
+        auto charged_particle_assocs = event->Get<edm4eic::MCRecoParticleAssociation>("ReconstructedChargedParticlesAssociations");
+        // TODO: NWB: Inconsistent plurals in collection names: "ReconstructedChargedParticlesAssociations" vs "EcalEndcapNClusterAssociations"
```

Comment:
We should change the name to "ReconstructedChargedParticleAssociations" now before people start using the inconsistent name in downstream code.

Reply:
I've cheerfully stayed out of any discussion about how things are named in EICrecon so far, but at some point I'd like to point out that "ReconstructedChargedParticles" is an input to `MatchClusters`, whereas "ReconstructedParticles" is an output. Maybe this would make perfect sense if I had more physics domain knowledge, but I would at least argue for "ReconstructedParticles" to be renamed to something like "MatchedReconstructedParticles" since we have a lot of collections of `edm4eic::ReconstructedParticle`s floating around and I can't figure out where they fit in the chain from how they are named.

Reply:
To be fair, it's probably `ReconstructedChargedParticles` that's the outlier here. It is (ab)using the structure that stores the fully-reconstructed particles as a container to pass track reconstruction results only. Could definitely be clearer indeed.